qemu/tests/block_hotplug.py

```diff
@@ -85,13 +87,13 @@ def get_new_disk(disk1, disk2):
             utils_test.run_virt_sub_test(test, params, env, sub_type)
 
         for num in xrange(blk_num):
-            device = qdevices.QDevice(pci_type)
             if params.get("need_plug") == "yes":
+                device = qdevices.QDevice(pci_type)
```

Comment:
why move it inside `if params.get("need_plug") == "yes"` ? it seems `params.get("need_plug") != "yes"` there will be problems for line 118

Reply:
For original design, "device = qdevices.QDevice(pci_type)" has two usage. One to define the unplugged device, the other one  to defined the new hot plugged device.

In my design, i only use it to define the new hot plugged device, and get the unplug device by other method, ex, -device id.

Reply:
BTW, line 118 belongs to codes under  if params.get("need_plug") == "yes".

Reply:
So, if params.get("need_plug") != "yes", code in line 118 will not be executed.

Reply:
got it， thanks